src/olympia/reviewers/tests/test_serializers.py

```diff
@@ -152,6 +155,94 @@ def test_get_entries_cached(self):
+        request.versioning_scheme = api_settings.DEFAULT_VERSIONING_CLASS()
+        request.version = api_version
+        extra_context.setdefault('request', request)
+
+        return FileEntriesDiffSerializer(
+            instance=obj, context=extra_context)
+
+    def serialize(self, obj, **extra_context):
+        return self.get_serializer(obj, **extra_context).data
+
+    def test_basic(self):
+        parent_version = self.addon.current_version
+
+        new_version = version_factory(
+            addon=self.addon, file_kw={
+                'filename': 'webextension_no_id.xpi',
+                'is_webextension': True,
+            }
+        )
+
+        repo = AddonGitRepository.extract_and_commit_from_version(new_version)
+
+        apply_changes(repo, new_version, 'Updated test file\n', 'test.txt')
+
+        file = self.addon.current_version.current_file
+
+        data = self.serialize(file, parent_version=parent_version)
+
+        assert data['id'] == file.pk
+        assert data['status'] == 'public'
+        assert data['hash'] == ''
+        assert data['is_webextension'] is True
+        assert data['created'] == (
+            file.created.replace(microsecond=0).isoformat() + 'Z')
+        assert data['url'] == (
+            'http://testserver/firefox/downloads/file/{}'
+            '/webextension_no_id.xpi?src=').format(file.pk)
+
+        assert data['selected_file'] == 'manifest.json'
+        assert data['download_url'] == absolutify(reverse(
+            'reviewers.download_git_file',
+            kwargs={
+                'version_id': self.addon.current_version.pk,
+                'filename': 'manifest.json'
+            }
+        ))
+
+        assert set(data['entries'].keys()) == {
+            'manifest.json', 'README.md', 'test.txt'}
+
+        manifest_data = data['entries']['manifest.json']
+        assert manifest_data['depth'] == 0
+        assert manifest_data['filename'] == u'manifest.json'
+        assert manifest_data['sha256'] == (
+            'bf9b0744c0011cad5caa55236951eda523f17676e91353a64a32353eac798631')
+        assert manifest_data['mimetype'] == 'application/json'
+        assert manifest_data['mime_category'] == 'text'
+        assert manifest_data['path'] == u'manifest.json'
+        assert manifest_data['size'] == 621
+        assert manifest_data['status'] == ''
+        assert isinstance(manifest_data['modified'], datetime)
+
+        readme_data = data['entries']['test.txt']
+        assert readme_data['depth'] == 0
+        assert readme_data['filename'] == u'test.txt'
+        assert readme_data['sha256'] == (
+            'f8b40fc302692ea4f552cb3d60bc89dd8b4616e398de5585e471cee73e2c0618')
+        assert readme_data['mimetype'] == 'text/plain'
+        assert readme_data['mime_category'] == 'text'
+        assert readme_data['path'] == u'test.txt'
+        assert readme_data['size'] == 18
+        assert readme_data['status'] == 'A'
```

Comment:
I don't see a test for showing deleted files. Is there one? I'm a bit confused about how we will access deleted files because it doesn't really make sense to include deleted files in `data['entries']`. Or does it?

Some details: https://github.com/mozilla/addons-code-manager/issues/413#issuecomment-479659974

Reply:
You're right, I added proper support for file deletion in 2367135ea9.

So, deleted files are in `data['entries']`, only they have their status `'D'` for deleted and most of their properties set to `null` given that we don't care anymore (and would have to fetch the previous commit to fetch the data which is unnecessary)

This is the first try at this, very happy to hear feedback regarding structure and data. For me, it'd be easier to just have the status in the end-structure but I figured it might be easier for you to reuse the same data-structure.

Does that work for you like that?

Reply:
Yep, this looks good: https://github.com/mozilla/addons-server/commit/2367135ea975e0f0aec0ff4c8f4f8927de0d63ae#diff-b02a60627dc42d205bfd52770cafe54cR250

Thanks!